bcda/main.go

```diff
@@ -610,7 +613,9 @@ func cleanupArchive(hrThreshold int) error {
 }
 
 func createAlphaEntities(acoSize string) (aco models.ACO, err error) {
-	tx := database.GetGORMDbConnection().Begin()
+	db := database.GetGORMDbConnection()
+	defer database.Close(db)
```

Comment:
This function already has a defer on the transaction. We can't close the connection before that transaction handling executes.

Reply:
Defers execute in a last in first out order so as I don't think this is an issue.  The other defer will execute first and then the db connection will close.

Reply:
we should be good to go https://tour.golang.org/flowcontrol/13